app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/event_screen/EventScreenEvent.kt

```diff
@@ -86,11 +95,26 @@ sealed interface EventScreenEvent {
         sealed interface DateTimePayload {
             val dateTime: ZonedDateTime
         }
+        sealed interface PhotoPayload {
+            val photo: Photo
+        }
+        sealed interface PhotoIdPayload {
+            val photoId: PhotoId
+        }
+        sealed interface AttendeePayload {
+            val attendee: Attendee
+        }
+        sealed interface AttendeeIdPayload {
+            val attendeeId: AttendeeId
+        }
```

Comment:
I think I asked this before and maybe I've missed the answer, but what's the benefit of wrapping these in sealed interfaces?

Reply:
When the event is received and processed, it will have a specific type associated with it, and not a general type. Make I can associate a Generic type with it, but it allows for easy "decompose" of the type of data, like Im doing here:

https://github.com/realityexpander/Tasky/blob/134ccc3d41384e7ab4054cafaba8956e39f94fe0/app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/event_screen/EventScreen.kt#L805

Reply:
I made a video explaining this functionality:

https://youtu.be/mqlVTJw0avY

Reply:
https://www.loom.com/share/aa8568f1b85142a194b7aa7d65265d88